applications/RANSApplication/custom_utilities/rans_nut_utility.cpp

```diff
@@ -0,0 +1,241 @@
+
+// Include base h
+#include "rans_nut_utility.h"
+
+namespace Kratos
+{
+RansNutUtility::RansNutUtility(
+    ModelPart& rModelPart,
+    const Variable<double>& rTurbulenceVariable1,
+    const Variable<double>& rTurbulenceVariable2,
+    const double RelativeTolerance,
+    const double AbsoluteTolerance,
+    const int EchoLevel) :
+    mrModelPart(rModelPart),
+    mEchoLevel(EchoLevel),
+    mRelativeTolerance(RelativeTolerance),
+    mAbsoluteTolerance(AbsoluteTolerance),
+    mrTurbulenceVariable1(rTurbulenceVariable1),
+    mrTurbulenceVariable2(rTurbulenceVariable2)
+{
+    KRATOS_TRY
+
+    block_for_each(mrModelPart.Elements(),[&](const ElementType& rElement) {
+        KRATOS_ERROR_IF_NOT(rElement.Has(CONSTITUTIVE_LAW)) << "CONSTITUTIVE_LAW is not found in element"
+            <<" data value container. [ Element.Id() = " << rElement.Id() << " ].\n";
+
+        const auto p_constitutive = rElement.GetValue(CONSTITUTIVE_LAW);
+        const auto rans_cl_name = p_constitutive->Info();
+
+        KRATOS_ERROR_IF(rans_cl_name.substr(0, 4) != "Rans")
+            << "Incompatible constitutive law is used. Please use constitutive "
+               "laws which starts with \"Rans*\" [ Constitutive law "
+               "name = "
+            << rans_cl_name << ", Element.Id() = " << rElement.Id() << " ].\n";
+    });
+
+    KRATOS_CATCH("");
+}
+
+void RansNutUtility::Initialize()
+{
+    KRATOS_TRY
+
+    // In here we transfer turbulence model variables to non-historical
+    // data value container of nodes so, nu_t can be calculated on those values
+    // this is done in order to achieve easier convergence.
+
+    block_for_each(mrModelPart.Nodes(), [&](NodeType& rNode){
+        rNode.SetValue(mrTurbulenceVariable1, rNode.FastGetSolutionStepValue(mrTurbulenceVariable1));
+        rNode.SetValue(mrTurbulenceVariable2, rNode.FastGetSolutionStepValue(mrTurbulenceVariable2));
+    });
+
+    // Initialize elemental turbulent viscosity also as the initial values
+    const auto& r_process_info = mrModelPart.GetProcessInfo();
+    block_for_each(mrModelPart.Elements(), TLSType(), [&](ElementType& rElement, TLSType& rTLS){
+        rElement.SetValue(TURBULENT_VISCOSITY, CalculateTurbulentViscosity(rElement, rTLS, r_process_info));
+    });
+
+    KRATOS_INFO_IF(this->Info(), mEchoLevel > 0)
+        << "Following variables are properly initialized on " << mrModelPart.Name() << ":\n"
+        << "       " << mrTurbulenceVariable1.Name() << "\n"
+        << "       " << mrTurbulenceVariable2.Name() << "\n"
+        << "       " << TURBULENT_VISCOSITY.Name() << "\n";
+
+    KRATOS_CATCH("");
+}
+
+void RansNutUtility::InitializeCalculation()
+{
+    KRATOS_TRY
+
+    const auto& r_process_info = mrModelPart.GetProcessInfo();
+    const auto& elements = mrModelPart.Elements();
+    const int number_of_elements = elements.size();
+
+    if (static_cast<int>(mElementData.size()) < number_of_elements) {
```

Comment:
you could declare `number_of_elements` as `std::size_t` and get rid of the `static_cast` here.